collection/slice_test.go

```diff
@@ -0,0 +1,160 @@
+package collection_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wego/pkg/collection"
+)
+
+func Test_Chunk(t *testing.T) {
```

Comment:
should we separate the test cases by type in its own function?

Reply:
We can do it. But is it necessary?
As of now, the test case will look like this:

```
--- FAIL: Test_Chunk/string_slice_has_more_items_than_chunk_size (0.00s)
        slice_test.go:91: 
                Error Trace:    slice_test.go:91
                Error:          Not equal: 
                                expected: [][]string{[]string{"1", "2"}, []string{"0", "4"}, []string{"5"}}
                                actual  : [][]string{[]string{"1", "2"}, []string{"3", "4"}, []string{"5"}}
```

I think it's clear enough.